test/ecto/adapters/postgres_test.exs

```diff
@@ -769,6 +770,12 @@ defmodule Ecto.Adapters.PostgresTest do
 
     query = update("prefix", "schema", [:x, :y], [id: 1, updated_at: nil], [])
     assert query == ~s{UPDATE "prefix"."schema" SET "x" = $1, "y" = $2 WHERE "id" = $3 AND "updated_at" IS NULL}
+
+    query = from(m in Schema, update: [set: [x: 0, y: 1]], where: [id: 0]) |> plan(:update_all) |> update()
+    assert query == ~s{UPDATE "schema" AS s0 SET "x" = 0, "y" = 1 WHERE (s0."id" = 0)}
+
+    query = from(m in Schema, update: [set: [x: 0, y: 1]], where: [id: 0], select: [:id]) |> plan(:update_all) |> update()
+    assert query == ~s{UPDATE "schema" AS s0 SET "x" = 0, "y" = 1 WHERE (s0."id" = 0) RETURNING s0."id"}
```

Comment:
@josevalim it would be greatly appreciated if you could help to take a look at this. I wonder if this return is the desired outcome of the function, or should we keep it the same as the previous one?